File: src/SalonOS.Shared/TenantEntity.cs
namespace SalonOS.Shared;

public abstract class TenantEntity
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid TenantId { get; set; }

    // ── Audit columns ──────────────────────────────────────────
    public DateTime? UpdatedAt { get; set; }

    // ── Soft-delete ─────────────────────────────────────────────
    public bool IsDeleted { get; set; } = false;
    public DateTime? DeletedAt { get; set; }
}

File: smart_salon_app/lib/domain/entities/tenant_entity.dart
import '../../types.dart';
import 'base_entity.dart';

abstract class TenantEntity extends BaseEntity {
  final String tenantId;

  const TenantEntity({
    required super.id,
    required this.tenantId,
    super.createdAt,
    super.updatedAt,
  });
}
